sdk/storage/azure-storage-blob/src/test/java/com/azure/storage/blob/specialized/BlobBaseAPITest.groovy

```diff
@@ -0,0 +1,602 @@
+
+        /* The QQ Avro stream has the following pattern
+           n * (data record -> progress record) -> end record */
+        // 1KB of data will only come back as a single data record.
+        /* Pretend to read more data because the input stream will not parse records following the data record if it
+         doesn't need to. */
+        readFromInputStream(qqStream, Constants.MB)
+
+        then:
+        // At least the size of blob to read will be in the progress list
+        mockReceiver.progressList.contains(sizeofBlobToRead)
+
+        /* Output Stream. */
+        when:
+        mockReceiver = new MockProgressReceiver()
+        options = new BlobQueryOptions()
+            .setProgressReceiver(mockReceiver as ProgressReceiver)
+        bc.queryWithResponse(new ByteArrayOutputStream(), expression, options, null, null)
+
+        then:
+        mockReceiver.progressList.contains(sizeofBlobToRead)
+    }
+
+    @Requires( { liveMode() } ) // Large amount of data.
+    def "Query multiple records with progress receiver"() {
+        setup:
+        BlobQueryDelimitedSerialization ser = new BlobQueryDelimitedSerialization()
+            .setRecordSeparator('\n' as char)
+            .setColumnSeparator(',' as char)
+            .setEscapeChar('\0' as char)
+            .setFieldQuote('\0' as char)
+            .setHeadersPresent(false)
+        uploadCsv(ser, 512000)
+
+        def mockReceiver = new MockProgressReceiver()
+        def expression = "SELECT * from BlobStorage"
+        BlobQueryOptions options = new BlobQueryOptions()
+            .setProgressReceiver(mockReceiver as ProgressReceiver)
+
+        /* Input Stream. */
+        when:
+        InputStream qqStream = bc.openQueryInputStream(expression, options)
+
+        /* The Avro stream has the following pattern
+           n * (data record -> progress record) -> end record */
+        // 1KB of data will only come back as a single data record.
+        /* Pretend to read more data because the input stream will not parse records following the data record if it
+         doesn't need to. */
+        readFromInputStream(qqStream, 16 * Constants.MB)
+
+        then:
+        long temp = 0
+        // Make sure theyre all increasingly bigger
+        for (long progress : mockReceiver.progressList) {
+            assert progress >= temp
+            temp = progress
+        }
+
+        /* Output Stream. */
+        when:
+        mockReceiver = new MockProgressReceiver()
+        temp = 0
+        options = new BlobQueryOptions()
+            .setProgressReceiver(mockReceiver as ProgressReceiver)
+        bc.queryWithResponse(new ByteArrayOutputStream(), expression, options, null, null)
+
+        then:
+        // Make sure theyre all increasingly bigger
+        for (long progress : mockReceiver.progressList) {
+            assert progress >= temp
+            temp = progress
+        }
+    }
+
+    class MockProgressReceiver implements ProgressReceiver {
```

Comment:
I'd move these classes to the bottom of the files so that we don't mix tests with test aids.

Reply:
done